src/platform/posix/split.rs

```diff
@@ -106,12 +133,43 @@ pub struct Writer {
     pub(crate) fd: Arc<Fd>,
     pub(crate) offset: usize,
     pub(crate) buf: Vec<u8>,
+    pub(crate) mtu: u16,
 }
 
 impl Writer {
     pub(crate) fn set_mtu(&mut self, value: u16) {
+        self.mtu = value;
         self.buf.resize(value as usize + self.offset, 0);
     }
+
+    pub(crate) fn send(&self, buf: &[u8]) -> io::Result<usize> {
+        let mut stack_buf = [0u8; crate::DEFAULT_MTU as usize + PIL];
```

Comment:
See above. We still can compare `crate::DEFAULT_MTU as usize + PIL` and `len` first, then determine whether we need `stack_buf`.

Reply:
![QQ_1721700667141](https://github.com/user-attachments/assets/41cbab0b-5481-4b38-8bcd-eda4174567e5)
![QQ_1721701049208](https://github.com/user-attachments/assets/2e96ba1b-4174-46da-a036-681d83a6ef2e)